### ISSUE TITLE
clarify that exists() is an extension to ERC-721

### ISSUE THREAD

**Issue by fulldecent** [OpenZeppelin/openzeppelin-solidity#1192]
This statement is incorrect:

https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC721/DeprecatedERC721.sol#L9

And this is a proprietary extension to ERC-721. 

https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC721/ERC721BasicToken.sol#L69

Please add a note that the `exists` function is an extension and is not part of the standard. This will help avoid confusion.

**Comment by nventuro** [comment]
Hi @fulldecent!

The sentence on `DeprecatedERC721` is actually correct: it indicates that `ERC721` is the interface to use when interacting with standard-compliant ERC721 contracts (as opposed to `DeprecatedERC721`, which implements an older version of the standard). You can find our 1.12 ERC721 interface [here](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/release-v1.12.0/contracts/token/ERC721/ERC721.sol#L39). Please let us know if you find any differences between that contract and the standard!

From what I can tell, you seem to be right about the `exists` function not being part of the standard (at least according to [this document](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md), which I believe is the latest and finalized version of it). Thanks for reporting this!

I see two main courses of action that we can take regarding said function:
 1. Add a comment (as suggested by @fulldecent)
 2. Make `exists` `internal` (it is used in some `require` statements)

I'm leaning towards 2, since we're doing some API breakage in 2.0 anyway, and I'd rather the stable interface be as minimal as possible allowing us to add on top of it, rather than having to remove stuff in the future. @shrugs @frangio thoughts?

**Comment by nventuro** [comment]
Hmm, I missed `exists` being in `ERC721Basic`, and therefore also in `ERC721`. We definitely should remove it from there.

Is that method useful enough to justify providing a derived contract with that single addition? I'd prefer keeping it `internal`, giving our users the choice to expose it, instead of 'suggesting' said usage by having it be a thing.

**Comment by saurabh-lodha-16** [comment]
Hey @nventuro / @shrugs / @fulldecent 
I found this thread when I wanted to expose the _exists() function in a smart contract I am building.
From what I understand that this function is not part of the EIP and therefore removed from the implementation right?

Are there any issues if I expose this function? I want to expose it so that the UI can check if the particular NFT was minted or not. Sorry to ask this here, but I am new to solidity and did not find any other resource.

This is the only function I exposed - 
```
    function exists(uint256 _tokenId) external view returns (bool) {
        return _exists(_tokenId);
    }
```
Would really appreciate knowing if this works or if I am doing something wrong. Thanks
